src/test/regress/expected/merge.out

```diff
@@ -1274,7 +2350,54 @@ MERGE INTO t1
 		UPDATE SET val = t1.val + 1
 	WHEN NOT MATCHED THEN
 		INSERT (id, val) VALUES (s1.id, s1.val);
-ERROR:  MERGE command is not supported on distributed/reference tables yet
+ERROR:  MERGE command is not supported with combination of distributed/local tables yet
```

Comment:
are you familiar with arbitrary config tests? Has a README as well: https://github.com/citusdata/citus/tree/main/src/test/regress/citus_tests/arbitrary_configs


Would be great to have few basic tests in arbitrary configs, it ensures that there is no gap on any different configurations (e.g., MX vs non-MX, local vs remote execution, different shard counts, different executor/planner settings etc.)

Reply:
Added couple of scenarios here in `merge_arbitrary`. Will try to add some more tomorrow.

Reply:
Does CI trigger arbitrary config tests?

Reply:
Added Citus local tables test too, please let me know if you have any MX specific scenarios.